engine-rest/engine-rest/src/main/java/org/operaton/bpm/engine/rest/impl/FetchAndLockHandlerImpl.java

```diff
@@ -357,6 +366,27 @@ protected void parseUniqueWorkerRequestParam(String uniqueWorkerRequestParam) {
     }
   }
 
+  protected void initializeQueue(int capacity) {
+    LOG.log(Level.FINEST, "Initializing queue with capacity [{0}]", capacity);
+    queue = new ArrayBlockingQueue<>(capacity);
+  }
+
+  private static int parseBlockingQueueCapacityParam(String queueSizeRequestParam) {
+    int capacity = DEFAULT_BLOCKING_QUEUE_CAPACITY;
+    if (queueSizeRequestParam != null) {
+      try {
+        final int parsedCapacity = Integer.parseInt(queueSizeRequestParam);
+        if (parsedCapacity <= 0) {
+          throw new NumberFormatException("Parameter " + BLOCKING_QUEUE_CAPACITY_PARAM_NAME + " has to be greater than zero");
```

Comment:
I'm not the biggest fan of throwing something and catching it immediately in the same block. I'd argue this is not really a NumberFormatException either, it is used as a vehicle for program flow here

Reply:
Yes, saw that too. It is to use the same catch as when parseInt would fail. I was thinking more of an IllegalArgumentException here.
Maybe the flow could be slightly improved here.